### PR TITLE
Add support for 'no-lru' tag to spec tests

### PR DESCRIPTION
**Pull request by gsoltis**
Port of spec changes from here: https://github.com/firebase/firebase-ios-sdk/pull/1410

### CODE REVIEW

File: packages/firestore/test/unit/specs/describe_spec.ts [hunk 1]
```diff
@@ -103,6 +126,12 @@ export function specTest(
         runner = it.only;
       } else if (!WEB_SPEC_TEST_FILTER(tags)) {
         runner = it.skip;
+      } else if (usePersistence && tags.indexOf('no-lru') !== -1) {
+        runner = (fullname: string, fn: () => {}) => {
+          const skipComment = comment || '';
+          // tslint:disable-next-line:no-console
+          console.log('skipping "' + fullName + '"\n\t' + skipComment);
```
Comment (schmidt-sebastian) [inline packages/firestore/test/unit/specs/describe_spec.ts]: This is a little different from how we handle other "skipped" tests. If you think it adds value to print the comment during the test run, you could just call `console.log` here and then use it.skip as your runner. Alternatively, you could rely on these comments as they show up in the test definition and just use `it.skip`.

File: packages/firestore/test/unit/specs/describe_spec.ts [hunk 2]
```diff
@@ -82,7 +85,27 @@ export function specTest(
...
+  comment: string,
+  builder: () => SpecBuilder
+): void;
+export function specTest(
+  name: string,
+  tags: string[],
+  commentOrBuilder: string | (() => SpecBuilder),
+  maybeBuilder?: () => SpecBuilder
 ): void {
+  let comment: string;
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply packages/firestore/test/unit/specs/describe_spec.ts]: This is intentionally `undefined`. It prevents adding a bunch of `comment: null` items to spec tests without comments.

File: packages/firestore/test/unit/specs/describe_spec.ts [hunk 3]
```diff
@@ -82,7 +85,27 @@ export function specTest(
...
+  let builder: () => SpecBuilder;
+  if (typeof commentOrBuilder === 'string') {
+    comment = commentOrBuilder;
+    builder = maybeBuilder;
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply packages/firestore/test/unit/specs/describe_spec.ts]: Added the `assert` below to catch both code paths.

File: packages/firestore/test/unit/specs/describe_spec.ts (hunk 1 above)
Comment (gsoltis) [reply packages/firestore/test/unit/specs/describe_spec.ts]: Done.

File: packages/firestore/test/unit/specs/describe_spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply packages/firestore/test/unit/specs/describe_spec.ts]: Even if `null` is not the best choice for this, you should still change the type here to `string|undefined`.

**Comment by gsoltis** [comment]
Done.

On Mon, Jun 18, 2018 at 1:40 PM, Sebastian Schmidt <notifications@github.com

**Pull request merged**